core/src/main/java/fr/sncf/osrd/utils/graph/GraphAdapter.java

```diff
@@ -1,24 +1,34 @@
 package fr.sncf.osrd.utils.graph;
 
-import com.google.common.graph.ImmutableNetwork;
+import fr.sncf.osrd.sim_infra.api.BlockInfra;
+import fr.sncf.osrd.sim_infra.api.RawSignalingInfra;
+import java.util.ArrayList;
 import java.util.Collection;
 
-/** Implements our custom Graph interface using an ImmutableNetwork */
-public class GraphAdapter<NodeT, EdgeT> implements Graph<NodeT, EdgeT> {
+public class GraphAdapter implements Graph<Integer, Integer> {
```

Comment:
Why are we not using an ImmutableGraph anymore like the LegacyGraphAdapter did ? And do we want to lose that ?

Reply:
Seen with @multun, all good :)